examples/adaptive_library/OblongStatorDuct.py

```diff
@@ -0,0 +1,372 @@
+            # end point is on symmetry boundary of geometry
+            center_t = end_piont_t
+            new_start_x, new_start_y = entity_start.x, entity_start.y
+            new_end_x, new_end_y = rt_to_xy(end_point_r + height, end_piont_t)
+
+    center_x, center_y = rt_to_xy(center_r, center_t)
+    center = Coordinate(center_x, center_y)  # center point
+    new_start_point = Coordinate(new_start_x, new_start_y)
+    new_end_point = Coordinate(new_end_x, new_end_y)
+    return r, center, new_start_point, new_end_point
+
+
+# %%
+# Get required parameters and objects
+# -----------------------------------
+# From Motor-CAD, get the adaptive parameters and their values.
+#
+# Use the ``set_default_parameter()`` method to set the required ``Duct Arc Height`` parameter
+# if undefined.
+set_default_parameter("Duct Arc Height", 0.7)
+
+
+# %%
+# Set required parameters for the oblong: height of arc ``Duct Arc Height``, duct width and duct
+# height.
+duct_arc_height = mc.get_adaptive_parameter_value("Duct Arc Height")
+duct_height = mc.get_variable("CircularDuctL1ChannelHeight")
+duct_width = mc.get_variable("CircularDuctL1ChannelWidth")
+duct_area = duct_height * duct_width
+
+# %%
+# Get the standard template stator region. This can be drawn for debugging if required.
+st_region = mc.get_region("stator")  # get the stator region
+
+# %%
+# Create the Adaptive Templates geometry
+# --------------------------------------
+# For each child region of the stator region:
+#
+# * Check whether the region is a stator duct.
+#
+# * Find the top and bottom lines that makes up the duct.
+#
+# * Modify the lines with respective arcs.
+#
+# * Set the region in Motor-CAD.
+#
+# The script accounts for whether ducts are full ducts or half ducts (the case when a duct spans the
+# rotor pole boundary)
+
+for child_name in st_region.child_names:
+    if "StatorDuctFluidRegion" in child_name:
+        duct_region = mc.get_region(child_name)
+        if round(duct_region.area / duct_area, 2) == 1:  # check if  full duct is drawn
+            for i, entity in enumerate(duct_region.entities):
+                if round(entity.length / duct_width, 2) == 1:  # check if the line is duct width
+                    # additional check in case width = height
+                    _, angle_start_point_angle = xy_to_rt(entity.start.x, entity.start.y)
+                    _, angle_end_point_angle = xy_to_rt(entity.end.x, entity.end.y)
+                    if (
+                        abs(angle_end_point_angle - angle_start_point_angle) > 0.05
+                    ):  # 0.05 degree is tolerance
+                        # get radius and center
+                        # convert this line segment to Arc
+                        Line_origin = check_line_origin_distance(
+                            i, duct_region
+                        )  # line near of far from origin
+                        radius, center = get_arc_radius_center(
```

Comment:
This can be updated for new Arc definition - we only need a centre or a radius now. Pass these as optional args e.g.
```Python
 Arc(entity.start, entity.end, radius=arc_radius)
```